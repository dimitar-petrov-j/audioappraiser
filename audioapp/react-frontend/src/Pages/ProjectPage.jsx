import React, { useState, useEffect} from "react";
import "../App.css";
import ProjectService from "../services/ProjectService.js";
import "bootstrap/dist/css/bootstrap.min.css";
import Jumbotron from "react-bootstrap/Jumbotron";
import Button from "react-bootstrap/Button";



function ProjectPage(){

    const [album, readyAlbum] = useState([]);

    useEffect(() => {

        //take value of id
        const value = window.location.href[window.location.href.length-1];

        console.log(value);


        ProjectService.getProjectById(value).then((response) => {
            readyAlbum(response.data);
        });

    }, []);

    function handleAmazonComSearch(){
        const url =
        "https://www.amazon.com/s?k=" +
        album.artist.replace(/ /g, "+") +
        "+" +
        album.name.replace(/ /g, "+") +
        "&i=popular";
        window.location.href = url;
    }

    function handleAmazonNLSearch(){
        const url =
        "https://www.amazon.nl/s?k=" +
        album.artist.replace(/ /g, "+") +
        "+" +
        album.name.replace(/ /g, "+") +
        "&i=popular";
        window.location.href = url;
    }


    function handleEbayComSearch(){
        const url =
        "https://www.ebay.com/sch/i.html?_from=R40&_trksid=p2334524.m570.l1311&_nkw=" +
        album.artist.replace(/ /g, "+") +
        "+" +
        album.name.replace(/ /g, "+") +
        "&_sacat=0&LH_TitleDesc=0&_osacat=0&_odkw=" +
        album.artist.replace(/ /g, "+") +
        "+" +
        album.name.replace(/ /g, "+");
        window.location.href = url;
    }


    return(
        <Jumbotron className="jumbotron-albumpage">
            <img className="img-albumpage-wrapper" src="https://e.snmc.io/i/fullres/s/ead2038df1155061252970b1c96bf24f/7125159"/><br/>
            <div className="right-albumpage">
                <p>Name: {album.name}</p>
                <p>Artist: {album.artist}</p>
                <p>Score: {album.rating}</p>
                <Button onClick={handleAmazonComSearch} className="button-albumpage">Find in amazon.com</Button>
                <Button onClick={handleAmazonNLSearch} className="button-albumpage">Find in amazon.nl</Button>
                <Button onClick={handleEbayComSearch} className="button-albumpage">Find in ebay.com</Button>
            </div>

        </Jumbotron>
    );
}

export default ProjectPage;