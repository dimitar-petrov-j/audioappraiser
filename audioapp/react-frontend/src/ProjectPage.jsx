import React, { useState, useEffect} from "react";
import "./App.css";
import ProjectService from "./services/ProjectService.js";
import "bootstrap/dist/css/bootstrap.min.css";
import Jumbotron from "react-bootstrap/Jumbotron";



function ProjectPage(){

    const [album, readyAlbum] = useState([]);

    useEffect(() => {

        //take value of id
        const value = window.location.href[window.location.href.length-1];

        console.log(value);


        ProjectService.getProjectById(value).then((response) => {
            readyAlbum(response.data);
        });

    }, []);

    return(
        <Jumbotron>
            <p>Name: {album.name}</p>
            <p>Artist: {album.artist}</p>
            <p>Score: {album.rating}</p>
        </Jumbotron>
    );
}

export default ProjectPage;