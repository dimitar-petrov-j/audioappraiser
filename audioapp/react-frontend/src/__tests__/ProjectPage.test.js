import React from "react"
import { cleanup, render, screen } from "@testing-library/react"
import ShallowRenderer from "react-test-renderer/shallow"
import ProjectPage from "../Pages/ProjectPage"


afterEach(cleanup);

it("renders correctly react-test-renderer", () => {
    const renderer = new ShallowRenderer()
    render(<ProjectPage />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
})