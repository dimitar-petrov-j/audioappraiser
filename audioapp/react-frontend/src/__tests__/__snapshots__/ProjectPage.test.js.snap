// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly react-test-renderer 1`] = `null`;
